solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -1183,6 +1191,18 @@ private void createEphemeralLiveNode() throws KeeperException, InterruptedExcept
     zkClient.multi(ops);
   }
 
+  private byte[] buildLiveNodeData() {
```

Comment:
wouldn't this belong in ZkStateReader, which is primarily responsible for reading/writing ZK?